File: Samples/Chapter04/GreetMe/FlagDlg.vb
Option Strict

Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Reflection
Imports System.Resources
Imports System.Globalization
Imports System.Threading

Namespace Apress.ExpertDotNet.GreetMeSample
	Public Class FlagDlg 
            Inherits Form
		Private FlagCtrl As PictureBox = New PictureBox()

		Public Sub New()
			Dim resManager As ResourceManager = New ResourceManager("Strings", [Assembly].GetExecutingAssembly())
			Me.Text = String.Format(resManager.GetString("DialogFormatString"), _
                resManager.GetString("DialogCaption"), _
                Thread.CurrentThread.CurrentUICulture.ToString())
			resManager = New ResourceManager("Flags", [Assembly].GetExecutingAssembly())
			Me.ClientSize = New Size(150,100)
			FlagCtrl.Image = DirectCast(resManager.GetObject("Flag"), Image)
			FlagCtrl.Location = New Point(20,20)
			FlagCtrl.Parent = Me
        End Sub
	End Class
End Namespace

File: Samples/Chapter04/GreetMe/FlagDlg.cs
using System;
//using System.Collections;
using System.ComponentModel;
using System.Drawing;
//using System.Data;
using System.Windows.Forms;
using System.Reflection;
//using System.Runtime.CompilerServices;
using System.Resources;
using System.Globalization;
using System.Threading;


namespace Apress.ExpertDotNet.GreetMeSample
{
	public class FlagDlg : Form
	{
		private PictureBox flagCtrl = new PictureBox();

		public FlagDlg()
		{
			ResourceManager resManager = new ResourceManager("Strings", Assembly.GetExecutingAssembly());
			this.Text = resManager.GetString("DialogCaption") + " " + Thread.CurrentThread.CurrentUICulture.ToString();
			resManager = new ResourceManager("Flags", Assembly.GetExecutingAssembly());
			this.ClientSize = new Size(150,100);
			flagCtrl.Image = (Image)resManager.GetObject("Flag");
			flagCtrl.Location = new Point(20,20);
			flagCtrl.Parent = this;
		}

	}
}